controllers/mirrorpeer_controller.go

```diff
@@ -288,43 +291,30 @@ func (r *MirrorPeerReconciler) deleteResources(mirrorPeer multiclusterv1alpha1.M
 			logger.Error(err, "Cannnot delete ManagedClusterAddon Resource", "ManagedClusterAddon", managedClusterAddOn)
 			return err
 		}
-		sourceSecrets, err := fetchAllSourceSecrets(context.TODO(), r.Client, mirrorPeer.Spec.Items[i].ClusterName)
-		if err != nil {
-			logger.Error(err, "Error in fetching Source Secrets")
-			return err
+
+		secretLabels := []string{string(utils.SourceLabel), string(utils.DestinationLabel)}
+
+		if mirrorPeer.Spec.ManageS3 {
+			secretLabels = append(secretLabels, string(utils.InternalLabel))
 		}
-		for _, sourceSecret := range sourceSecrets {
-			if err := r.Client.Delete(context.TODO(), &sourceSecret); err != nil {
-				logger.Info("Cannot delete Source Secret", "Source Secret", sourceSecret)
-				return err
-			}
+
+		secretRequirement, err := labels.NewRequirement(utils.SecretLabelTypeKey, selection.In, secretLabels)
+		if err != nil {
+			logger.Error(err, "cannot parse new requirement")
 		}
-	}
-	destination, err := fetchAllDestinationSecrets(context.TODO(), r.Client, "")
-	if err != nil {
-		logger.Info("Error in fetching Destination Secrets")
-		return err
-	}
-	for _, destinationSecret := range destination {
-		logger.Info(destinationSecret.Name)
-		if err := r.Client.Delete(context.TODO(), &destinationSecret); err != nil {
-			logger.Info("cannot delete destination secret")
-			return err
+
+		secretSelector := labels.NewSelector().Add(*secretRequirement)
+
+		deleteOpt := client.DeleteAllOfOptions{
+			ListOptions: client.ListOptions{
+				Namespace:     mirrorPeer.Spec.Items[i].ClusterName,
+				LabelSelector: secretSelector,
+			},
 		}
-	}
 
-	if mirrorPeer.Spec.ManageS3 {
-		for _, peerRef := range mirrorPeer.Spec.Items {
-			s3Secret := corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      utils.CreateUniqueSecretName(peerRef.ClusterName, peerRef.StorageClusterRef.Namespace, peerRef.StorageClusterRef.Name, utils.S3ProfilePrefix),
-					Namespace: peerRef.ClusterName,
-				},
-			}
-			logger.Info(s3Secret.ObjectMeta.Name)
-			if err := r.Client.Delete(context.TODO(), &s3Secret); err != nil {
-				return err
-			}
+		var secret corev1.Secret
+		if err := r.DeleteAllOf(context.TODO(), &secret, &deleteOpt); err != nil {
```

Comment:
this might delete everything even when only 1 mirrorpeer is deleted. did you verify that?

Reply:
updated with fix.